tests/unit/test_coupling.py

```diff
@@ -82,69 +93,122 @@ def test_real_output():
 
 
 @pytest.mark.basic
-def test_closest_tune_approach():
-    beam = 1
-    df_twiss = tfs.read(INPUT / "coupling_bump" / f"twiss.lhc.b{beam:d}.coupling_bump.tfs", index=NAME)
+@pytest.mark.parametrize(
+    "cta_method, max_relative_error_to_teapot, result_should_be_real",
+    [
+        ("calaga", 0, True),  # this is the same as teapot, hence 0 relative error
+        ("franchi", 0.001, True),
+        ("teapot_franchi", 0.0005, True),
+        ("persson", 0.25, False),  # not sure why it is so high from here
+        ("persson_alt", 0.25, False),
+        ("hoydalsvik", 0.25, False),
+        ("hoydalsvik_alt", 0.25, False),
+    ],
+)
+def test_closest_tune_approach(
+    cta_method, max_relative_error_to_teapot, result_should_be_real, _coupling_bump_teapot_cta
+):
+    df_twiss = tfs.read(COUPLING_BUMP_TWISS_BEAM_1, index=NAME)
     df = prepare_twiss_dataframe(df_twiss=df_twiss, max_order=7)
     df_cmatrix = coupling_via_cmatrix(df)
-    df_twiss[COUPLING_RDTS] = df_cmatrix[COUPLING_RDTS]
+    df_twiss[F1001] = df_cmatrix[F1001]  # ignoring F1010 in this test as it is bigger than F1001
```

Comment:
that part could also go in a fixture that you could use in the teapot fixture

Reply:
that's why I wanted to use a fixture in the first place, so it doesn't load the file for eacch method ^^

Reply:
The test suite is done very fast anyway (7s on my laptop).

Considering that `df_cmatrix` is also needed later down in the test I think it's fine as it is.